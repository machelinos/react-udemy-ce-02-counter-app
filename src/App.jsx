import { PropTypes } from 'prop-types'

export const App = ({ title, subtitle }) => {
  return (
    <>
      <h1 data-testid="test-title">{title}</h1>
      <p>{subtitle}</p>
    </>
  )
}

App.propTypes = {
  title: PropTypes.string.isRequired,
  subtitle: PropTypes.string.isRequired,
}

App.defaultProps = {
  title: 'Default title',
  subtitle: 'subtitle',
}
