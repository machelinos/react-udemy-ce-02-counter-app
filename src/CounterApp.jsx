import PropTypes from 'prop-types'
import { useState } from 'react'

export const CounterApp = ({ value }) => {
  const [count, setCount] = useState(value)

  const handleAdd = () => {
    setCount(count + 1)
  }

  const handleSubstract = () => {
    setCount(count - 1)
  }

  const handleReset = () => {
    setCount(value)
  }

  return (
    <>
      <h1>Counter App</h1>
      <h2>{count}</h2>
      <button onClick={handleAdd}>+1</button>
      <button onClick={handleSubstract}>-1</button>
      <button onClick={handleReset}>Reset</button>
    </>
  )
}

CounterApp.propTypes = {
  value: PropTypes.number.isRequired,
}
