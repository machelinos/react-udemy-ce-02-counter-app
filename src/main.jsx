import React from 'react'
import ReactDOM from 'react-dom/client'
import './styles.css'
import { App } from './App'

ReactDOM.createRoot(document.getElementById('root')).render(
  <React.StrictMode>
    <App title="Title" subtitle="Subtitle" />
  </React.StrictMode>,
)
