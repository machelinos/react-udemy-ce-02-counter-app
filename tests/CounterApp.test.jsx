import { render, screen } from '@testing-library/react'
import { CounterApp } from '../src/CounterApp'

describe('<CounterApp /> tests', () => {
  const value = 0

  test('<CounterApp /> render should match the snapshot', () => {
    const { container } = render(<CounterApp value={value} />)
    expect(container).toMatchSnapshot()
  })

  test('<CounterApp /> render should render the initial value given as a prop', () => {
    render(<CounterApp value={value} />)
    expect(screen.getByRole('heading', { level: 2 }).innerHTML).toBe(
      String(value),
    )
  })
})
