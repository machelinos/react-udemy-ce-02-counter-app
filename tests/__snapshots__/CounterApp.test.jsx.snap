// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CounterApp /> tests <CounterApp /> render should match the snapshot 1`] = `
<div>
  <h1>
    Counter App
  </h1>
  <h2>
    0
  </h2>
  <button>
    +1
  </button>
  <button>
    -1
  </button>
  <button>
    Reset
  </button>
</div>
`;
